devel/happy/Portfile

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           haskell_stack 1.0
+
+name                happy
+version             1.19.12
+categories          devel haskell
+maintainers         openmaintainer
+license             GPL-3
+platforms           macosx
+homepage            https://github.com/simonmar/happy
+
+description         Happy is a parser generator for Haskell 98 (and later).
+long_description    ${description}. \
+            Happy is a parser generator system for Haskell, similar to \
+            the tool `yacc' for C. Like `yacc', it takes a file containing \
+            an annotated BNF specification of a grammar and produces a \
+            Haskell module containing a parser for the grammar.  Happy \
+            is flexible: you can have several Happy parsers in the same \
+            program, and several entry points to a single grammar. Happy \
+            can work in conjunction with a lexical analyser supplied by \
+            the user (either hand-written or generated by another program), \
+            or it can parse a stream of characters directly (but this \
+            isn't practical in most cases).
+
+master_sites        https://hackage.haskell.org/package/${name}-${version}
+
+conflicts           hs-happy
+
+checksums           rmd160  34fff4797b87496a0493960b784df96385d4d6c0 \
+                    sha256  fb9a23e41401711a3b288f93cf0a66db9f97da1ce32ec4fffea4b78a0daeb40f \
+                    size    183254
+
+# relative paths to ${prefix}
+set happy_datadir   share/${name}
+
+post-extract {
+    xinstall -o macports -m 0755 -d \
```

Comment:
Why do you need the `-o macports` option here? The macports user can actually be re-named by users and some users do not use MacPorts with root privileges (and thus don't have privilege separation), which will cause this to fail.

This should't be necessary anyway, stack_root is IIRC automatically created by the PortGroup already, and anything in destroot shouldn't be required until the destroot phase and can be created there with the default owner.

Reply:
I’ll delete this.

It’s a legacy from copy-and-paste from previous bad hackery before the `Paths_*.hs` approach for Cabal `data-files` paths.